src/room.js

```diff
@@ -0,0 +1,40 @@
+class Room {
+
+  constructor (roomname) {
+    this.roomname = roomname;
+    this.endpoints = {};
+  }
+  addEndpoint (endpointId) {
+    this.endpoints[endpointId] = {
+      username: '',
+      permissions: 'CHAT',
+      commsId: ''
+    };
+  }
+  updateUsername (endpointId, username) {
+    this.endpoints[endpointId].username = username;
```

Comment:
I think you need to use bracket syntax, if `username` is not already defined? So:
```
this.endpoints[endpointId].username = username
```

(Same for the other `update` functions)

@njsfield @esraajb I might be wrong about that? But I think it wasn't working for me when I used dot syntax like this elsewhere.